### PR TITLE
fix: download JRE for keytool dependency and allow generation of pfx keys

### LINKED ISSUE #114: Download JRE if user needs to generate PFX keys
Download `adoptium` JRE and unpack to the `SOLO_HOME_DIR` and use it from there.

### PR DESCRIPTION
**Pull request by leninmehedy**
## Description

This pull request changes the following:

* feat: allow fetching keytool dependency automatically for pfx keys
* feat: allow node keys to generate pfx keys
* doc: refresh README with latest console output
* use `semver` package for version parsing instead of "reinventing the wheel".
* Added more test coverage

It installs JRE at `~/.solo/bin/jre` and `keytool` will be available at: `~/.solo/bin/jre/bin/keytool`

### Related Issues

* Closes #114 

### TODO

- [x] Install JRE only when user wants to create pfx keys
- [x] Translate the bash script into javascript code so that keytool is invoked using ShellRunner.run(...)

### CODE REVIEW

**Review by jeromy-cannon — APPROVED** [review]

**Review by JeffreyDallas — DISMISSED** [review]

File: test/e2e/core/platform_installer_e2e.test.mjs [hunk 1]
```diff
@@ -196,7 +183,9 @@ describe('PackageInstallerE2E', () => {
   describe('copyPlatformConfigFiles', () => {
     it('should succeed to copy platform config files for node0', async () => {
       const podName = 'network-node0-0'
-      await installer.resetHapiDirectories(podName)
+      await k8.execContainer(podName, constants.ROOT_CONTAINER, ['bash', '-c', `rm -f ${constants.HEDERA_HAPI_PATH}/*.txt`])
+      await k8.execContainer(podName, constants.ROOT_CONTAINER, ['bash', '-c', `rm -f ${constants.HEDERA_HAPI_PATH}/*.xml`])
+      await k8.execContainer(podName, constants.ROOT_CONTAINER, ['bash', '-c', `rm -f ${constants.HEDERA_HAPI_PATH}/data/config/*.properties`])
```
Comment (jeromy-cannon) [inline test/e2e/core/platform_installer_e2e.test.mjs:188]: This and the commands above will fail once I finish the configMap and secretMapping:
https://github.com/hashgraph/full-stack-testing/pull/797/files

configMap:
network-node-data-config-cm: /opt/hgcapp/services-hedera/HapiApp2.0/data/config
	• api-permission.properties
	• application.properties
	• bootstrap.properties
	• https://github.com/hashgraph/solo/issues/136
network-node-etc-cm: /etc/network-node
	• application.env
	• https://github.com/hashgraph/solo/issues/110
network-node-hapi-app-cm: /opt/hgcapp/services-hedera/HapiApp2.0
	• config.txt
	• settings.txt
	• log4j2.xml
	• settingsUsed.txt
	• https://github.com/hashgraph/solo/issues/136
network-node-platform-cm: /home/hedera/platform
	• build-${tag}.zip
	• build-${tag}.sha384
	• https://github.com/hashgraph/solo/issues/89

secrets:
network-node-hapi-app-secrets: /opt/hgcapp/services-hedera/HapiApp2.0/
	• hedera.crt
	• hedera.key
	• https://github.com/hashgraph/solo/issues/135
network-node-keys-secrets: /opt/hgcapp/services-hedera/HapiApp2.0/data/keys
	• a-private-node0.pem
	• a-public-node0.pem
	• a-public-node1.pem
	• a-public-node2.pem
	• s-private-node0.pem
	• s-public-node0.pem
	• s-public-node1.pem
	• s-public-node2.pem
	• https://github.com/hashgraph/solo/issues/135

File: test/e2e/core/platform_installer_e2e.test.mjs [hunk 2]
```diff
@@ -180,8 +168,7 @@ describe('PackageInstallerE2E', () => {
       fs.writeFileSync(path.join(keysDir, `hedera-${nodeId}.key`), '')
       fs.writeFileSync(path.join(keysDir, `hedera-${nodeId}.crt`), '')
 
-      await installer.resetHapiDirectories(podName)
-
+      await k8.execContainer(podName, constants.ROOT_CONTAINER, ['bash', '-c', `rm -f ${constants.HEDERA_HAPI_PATH}/hedera.*`])
```
Comment (jeromy-cannon) [inline test/e2e/core/platform_installer_e2e.test.mjs:171]: same as the note left on ../config/*.properties

File: test/e2e/core/platform_installer_e2e.test.mjs [hunk 3]
```diff
@@ -148,9 +136,9 @@ describe('PackageInstallerE2E', () => {
       const tmpDir = getTmpDir()
       const keysDir = path.join(tmpDir, 'keys')
       const shellRunner = new ShellRunner(testLogger)
-      await shellRunner.run(`test/scripts/gen-standard-keys.sh node0,node1 ${keysDir}`)
+      await shellRunner.run(`test/scripts/gen-openssl-keys.sh node0,node1 ${keysDir}`)
 
-      await installer.resetHapiDirectories(podName)
+      await k8.execContainer(podName, constants.ROOT_CONTAINER, ['bash', '-c', `rm -f ${constants.HEDERA_HAPI_PATH}/data/keys/*`])
```
Comment (jeromy-cannon) [inline test/e2e/core/platform_installer_e2e.test.mjs:141]: same as the note left on ../config/*.properties

File: test/e2e/core/platform_installer_e2e.test.mjs [hunk 4]
```diff
@@ -130,7 +118,7 @@ describe('PackageInstallerE2E', () => {
       const shellRunner = new ShellRunner(testLogger)
       await shellRunner.run(`test/scripts/gen-legacy-keys.sh node0,node1,node2 ${keysDir}`)
 
-      await installer.resetHapiDirectories(podName)
+      await k8.execContainer(podName, constants.ROOT_CONTAINER, ['bash', '-c', `rm -f ${constants.HEDERA_HAPI_PATH}/data/keys/*`])
```
Comment (jeromy-cannon) [inline test/e2e/core/platform_installer_e2e.test.mjs:121]: same as the note left on ../config/*.properties

**Review by jeromy-cannon — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Pull request merged**